src/server/string_family.cc

```diff
@@ -666,6 +666,91 @@ void StringFamily::GetSet(CmdArgList args, ConnectionContext* cntx) {
   return (*cntx)->SendNull();
 }
 
+void StringFamily::GetEx(CmdArgList args, ConnectionContext* cntx) {
+  string_view key = ArgS(args, 1);
+  string_view cur_val;
+
+  /*Get current value before set its expiration*/
+  auto cb_get = [&](Transaction* t, EngineShard* shard) {
+    return OpGet(t->GetOpArgs(shard), key, false);
+  };
+
+  DVLOG(1) << "Before Get::ScheduleSingleHopT " << key;
+
+  OpResult<string> result = cntx->transaction->ScheduleSingleHopT(std::move(cb_get));
+
+  if (result.status() != OpStatus::OK) {
+    switch (result.status()) {
+      case OpStatus::WRONG_TYPE:
+        (*cntx)->SendError(kWrongTypeErr);
+        break;
+      default:
+        DVLOG(1) << "GET " << key << " nil";
+        (*cntx)->SendNull();
+    }
+    return;
+  } else {
+    cur_val = result.value();
+  }
+
+  SetCmd::SetParams sparams;
+  int64_t int_arg = 0;
+
+  for (size_t i = 2; i < args.size(); i++) {
```

Comment:
unfortunately, this is not the right approach on how to implement commands in Dragonfly. 

This would break the command atomicity semantics because you have here 2 hops.
The right approach is to extend  `OpGet` functionality to support this use case and pass appropriate params there.

Please take a look at https://github.com/dragonflydb/dragonfly/pull/365 that had the same problem. Specifically:
1. Move argument parsing to the beginning of the function.
2. Introduce `ExpireParams` struct that consolidates all the expiry information. It should have a clear no-op/invalid state that means "no expiry info defined".
3. Extend OpGet to accept `ExpireParams` and optionally update expiry if it's defined. 

Some of the functionality above already exists in the codebase, so I took the liberty to introduce 
https://github.com/dragonflydb/dragonfly/pull/388 that should help you with implementing this PR. Please wait for it to be submitted and resync.

Reply:
Thanks a lot! Waiting for pr and then trying to fix the inappropriate implementation.

Reply:
The pr was merged, please rebase